pkg/operator/etcdmemberscontroller/etcdmemberscontroller.go

```diff
@@ -0,0 +1,230 @@
+package etcdmemberscontroller
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const workQueueKey = "key"
+
+// EtcdMembersController reports the status conditions
+// of etcd members.
+type EtcdMembersController struct {
+	operatorClient v1helpers.OperatorClient
+	etcdClient     etcdcli.EtcdClient
+
+	eventRecorder events.Recorder
+	queue         workqueue.RateLimitingInterface
+	cachesToSync  []cache.InformerSynced
+}
+
+func NewEtcdMembersController(operatorClient v1helpers.OperatorClient,
+	etcdClient etcdcli.EtcdClient,
+	kubeInformers operatorv1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+) *EtcdMembersController {
+	nodeInformer := kubeInformers.InformersFor("").Core().V1().Nodes()
+	endpointsInformer := kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints()
+	podInformer := kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods()
+	c := &EtcdMembersController{
+		operatorClient: operatorClient,
+		etcdClient:     etcdClient,
+
+		cachesToSync: []cache.InformerSynced{
+			operatorClient.Informer().HasSynced,
+			endpointsInformer.Informer().HasSynced,
+			nodeInformer.Informer().HasSynced,
+			podInformer.Informer().HasSynced,
+		},
+		eventRecorder: eventRecorder.WithComponentSuffix("member-observer-controller"),
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EtcdMembersController"),
+	}
+	operatorClient.Informer().AddEventHandler(c.eventHandler())
```

Comment:
too frequent.  We just want to trigger every minute. This will be way too noisy.

Reply:
added a time-based trigger.